tests/fs/test_trackingdelegate.out

```diff
@@ -4,8 +4,8 @@ Wrote to file "/file.txt"
 About to move "/file.txt" to "/renamed.txt"
 Moved "/file.txt" to "/renamed.txt"
 Opened "/renamed.txt" with flags 1
-Wrote to file "/dev/tty"
-File read returned 'hello!'
+File read returned 'hello!Wrote to file "/dev/tty"
+'
```

Comment:
What happened here?

Reply:
Yes, this one's a bit kooky.  Internally 'printf', for reasons known only to itself, splits its calls to 'write' into two, in this case for the first 26 characters followed by the remaining 2 chars.  So the delegate function is called after the first 26 characters have been written.  The original code buffers this though, so output from the delegate function is printed first, followed by the printf output.  With the changes it prints the first 26 characters of the printf string, followed by the delegate function output then the last two characters.

In practice I don't think this behaviour change is necessarily a problem, since in a real program, if the delegate functions are used, I'd imagine they probably wouldn't print anything.

Reply:
Oh gotcha, this looks fine then. Good sleuthing!